spec/services/coronavirus_pages/timeline_entry_builder_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+RSpec.describe CoronavirusPages::TimelineEntryBuilder do
+  let(:fixture_path) { Rails.root.join "spec/fixtures/coronavirus_landing_page.yml" }
+  let(:github_content) { YAML.safe_load(File.read(fixture_path)) }
+
+  let(:coronavirus_page) { create(:coronavirus_page, :landing) }
+
+  before do
+    stub_request(:get, Regexp.new(coronavirus_page.raw_content_url))
+      .to_return(status: 200, body: github_content.to_json)
+  end
+
+  it "creates new timeline entries from the yaml file" do
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.count).to eq(2)
+
+    first_entry = coronavirus_page.timeline_entries.first
+
+    expect(first_entry.heading).to eq(github_entry_list.first["heading"])
+    expect(first_entry.content).to eq(github_entry_list.first["paragraph"])
+  end
+
+  it "removes existing timeline entries before creating new ones" do
+    create(:timeline_entry, coronavirus_page: coronavirus_page)
+    create(:timeline_entry, coronavirus_page: coronavirus_page)
+    create(:timeline_entry, coronavirus_page: coronavirus_page)
+
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.count).to eq(2)
+  end
+
+  it "doesn't remove existing timeline entries if the YAML file is empty" do
+    github_content = YAML.safe_load(File.read(fixture_path))
+    github_content["content"]["timeline"]["list"].clear
+
+    stub_request(:get, Regexp.new(coronavirus_page.raw_content_url))
+      .to_return(status: 200, body: github_content.to_json)
+
+    entry = create(:timeline_entry, coronavirus_page: coronavirus_page)
+
+    described_class.new.create_timeline_entries
+
+    first_entry = coronavirus_page.timeline_entries.first
+
+    expect(coronavirus_page.timeline_entries.count).to eq(1)
+    expect(first_entry.heading).to eq(entry.heading)
+    expect(first_entry.content).to eq(entry.content)
+    expect(first_entry.created_at).to eq(entry.created_at)
+  end
+
+  it "keeps the timeline entries in the right order" do
+    new_data = { "heading" => "heading", "paragraph" => "paragraph" }
+
+    github_content = YAML.safe_load(File.read(fixture_path))
+    github_content["content"]["timeline"]["list"].unshift(new_data)
+
+    stub_request(:get, Regexp.new(coronavirus_page.raw_content_url))
+      .to_return(status: 200, body: github_content.to_json)
+
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.first.heading).to eq("heading")
+    expect(coronavirus_page.timeline_entries.first.position).to eq(3)
```

Comment:
I don't think this test is right as unshift inserts this at the start of the array - whereas this tests that the item has the last position.

Reply:
I've fixed this, it is expected to be able to add things to the start and have them written to the last position. That's so that the newest entries can be added to the top of the yaml whilst the oldest end up in position 1. The tests now reflect this.